profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReaderMapEntrySource.java

```diff
@@ -0,0 +1,7 @@
+package com.scottlogic.deg.profile.reader;
+
+import java.util.stream.Stream;
+
+public interface ConstraintReaderMapEntrySource {
+    Stream<ConstraintReaderMapEntry> getConstraintReaderMapEntries();
```

Comment:
Why are we using a stream over a collection here?

Reply:
No particular reason - which do you think would be better?

Reply:
I would've set a collection because we don't care about any of the aspects of a stream - we aren't using lazy evaluation, and we always evaluate all of the elements.